src/context/Context.tsx

```diff
@@ -1,54 +1,42 @@
 // @ts-ignore
 import React, {
-  useState,
   createContext,
   PropsWithChildren,
-  ReactNode
+  ReactNode,
+  useMemo,
+  useState,
 } from 'react'
-import WebViewer, { WebViewerInstance } from '@pdftron/webviewer'
+import type { WebViewerInstance } from '@pdftron/webviewer'
 
 export type TProviderProp = {
-  libLocation: string,
   children: ReactNode
 }
 
 export type TContextState = {
-  instances: object,
-  addInstance: Function
-}
-
-export type TKeyInstancePair = {
-  [key: string]: WebViewerInstance
+  instance: null | WebViewerInstance
+  setInstance(arg: WebViewerInstance): void
 }
 
 const DocumentViewerContext = createContext<TContextState>({
-  instances: {},
-  addInstance: async () => { },
+  instance: null,
+  setInstance: ()=>{}
 })
 
-function DocumentViewerProvider({ children, libLocation }: PropsWithChildren<TProviderProp>): JSX.Element {
-
-  const [instances, setInstances] = useState<TKeyInstancePair>({})
-  const value = { instances, addInstance }
-
-  async function addInstance(
-    initialDoc: string,
-    UID: string,
-    DVElement: HTMLElement
-  ) {
-    const instance = await WebViewer(
-      {
-        path: libLocation,
-        initialDoc: initialDoc,
-        disabledElements: ['header', 'toolsHeader', 'pageNavOverlay', 'textPopup', 'contextMenuPopup']
-      },
-      DVElement
-    )
-    setInstances({ ...instances, [UID]: instance })
+function DocumentViewerProvider({
+  children
+}: PropsWithChildren<TProviderProp>): JSX.Element { 
+
+  const [instanceState, setInstanceState] = useState<WebViewerInstance | null>(null)
+  const value = { instance: instanceState, setInstance }
+
+  function setInstance(instance: WebViewerInstance | null) {
+    setInstanceState(instance)
   }
 
+  const memoizedValue = useMemo(()=>{return value},[instanceState, setInstance])
```

Comment:
This is a very strange way to do this. You need to memoize the actual value object

```
const value = useMemo(()=> ({ instance: instanceState, setInstance }),[instanceState, setInstance])
```

Reply:
Now the actual object is memoized